assessment_utilities.py

```diff
@@ -669,21 +675,31 @@ def set_item_learning_objectives(data, form):
     return form
 
 def update_answer_form(answer, form, question=None):
-    if answer['type'] == 'answer-record-type%3Ashort-text-answer%40ODL.MIT.EDU':
+    if 'type' in answer:
+        if isinstance(answer['type'], list):
+            answer_types = answer['type']
+        else:
+            answer_types = [answer['type']]
+    elif 'recordTypeIds' in answer:
+        answer_types = answer['recordTypeIds']
+    else:
+        raise KeyError('missing answer type')
+
+    if 'answer-record-type%3Ashort-text-answer%40ODL.MIT.EDU' in answer_types:
```

Comment:
Of course we do :-) It's actually in the workplan to create at least one more.

I'm open to recommendations. Not sure what a better flow would be, since it depends on an out of band agreement between the client and server on what each question type can do...